### PR TITLE
fix: Handle blueprint schema data types without titles (as per CIP-57 titles are optional).

### PR DESCRIPTION
**Pull request by matiwinnetou**
## Summary

This PR enhances the blueprint annotation processor to properly handle schemas without titles, ensuring full CIP-57 compliance. When a schema lacks a title, the processor now extracts a valid class name from the definition key (e.g., `types/custom/Data` → `Data`), enabling successful code generation for blueprints like SundaeSwap V2 that use optional title fields.

Additionally, this PR improves error handling by introducing `BlueprintGenerationException` for consistent, actionable error reporting during annotation processing.

### 1. CIP-57 Compliance - Optional Title Handling

**Problem**: CIP-57 specifies that schema titles are optional, but the annotation processor was failing with NPE when encountering schemas without titles (e.g., SundaeSwap V2 blueprints).

**Solution**:
- Created `BlueprintUtil.getClassNameFromReferenceKey()` to extract class names from definition keys
- Added `FieldSpecProcessor.resolveTitleFromDefinitionKey()` to handle title resolution logic
- Falls back to definition key parsing when schema title is missing

**Files Modified**:
- `BlueprintUtil.java` - Added reference key parsing method
- `FieldSpecProcessor.java` - Extracted title resolution into testable method
- Added comprehensive Java comments explaining CIP-57 compliance

**Test Coverage**:
- `BlueprintUtilTest.GetClassNameFromReferenceKeyTests` - 9 unit tests
- `FieldSpecProcessorTest.ResolveTitleFromDefinitionKeyTests` - 13 unit tests
- `MissingTitleTest.shouldHandleMixedTitledAndUntitledDefinitionsWithGenerics()` - Integration test

### 2. Improved Error Handling

**Problem**: Blueprint processing errors were inconsistent - some threw `IllegalArgumentException`, others logged warnings and silently continued.

**Solution**:
- Created `BlueprintGenerationException` for all blueprint processing errors
- Replaced `IllegalArgumentException` with `BlueprintGenerationException` throughout annotation-processor module
- Added fail-fast behavior with clear, actionable error messages

**New Exception Class**:
```java
public class BlueprintGenerationException extends RuntimeException {
    public static BlueprintGenerationException forDefinition(String definitionKey, String message) {
        return new BlueprintGenerationException(
            String.format("Cannot process blueprint definition '%s': %s", definitionKey, message)
        );
    }
}
```

**Files Modified**:
- `BlueprintGenerationException.java` - NEW exception class
- `DatumModelFactory.java` - Schema title validation
- `SchemaTypeResolver.java` - Option/Pair type validation (4 throws)
- `FieldSpecProcessor.java` - Updated catch blocks
- `ValidatorProcessor.java` - Updated catch blocks
- `BlueprintAnnotationProcessor.java` - Added exception handling

**Intentionally Not Changed**:
- `ConverterCodeGenerator.java` - Generates runtime validation code (not annotation processing errors)

### 3. Test Improvements

**Refactored Tests**: Converted inline source strings to external resource files following established patterns:
- `MissingTitleTest.java` - Now uses `MissingTitleBlueprint.java`, `PrimitiveNoTitleBlueprint.java`
- `BlueprintAnnotationProcessorGenerationTest.GenericTypeSkipTests` - 4 tests now use external files

**Renamed Tests**: Changed project-specific names to problem-focused names:
- `SundaeSwapV2Pattern.java` → `MixedTitlesWithGenerics.java`
- `sundaeswap-v2-pattern_aiken_v1_0_26.json` → `mixed-titles-with-generics_aiken_v1_0_26.json`
- Updated documentation to emphasize CIP-57 compliance over project names

### Before:
```
IllegalArgumentException: Schema title cannot be null or empty
```

### After:
```
BlueprintGenerationException: Schema title cannot be null or empty.
Per CIP-57, schemas must have a title for class generation.
```

### Before:
```
IllegalArgumentException: Option type should have 2 anyOfs
```

### After:
```
BlueprintGenerationException: Invalid Option type schema: must have exactly 2
anyOf alternatives (None and Some). Found: 3
```

## Testing

✅ **All 213 tests passing**

### New Test Coverage:
- **Unit Tests**: 22 new tests for title resolution and reference key parsing
- **Integration Tests**: Validates mixed titled/untitled schemas with generics

### Key Test Cases:
- Schemas without titles (fallback to definition key)
- Mixed titled and untitled definitions
- Generic type instantiations (skipped correctly)
- JSON Pointer unescaping (`types~1custom~1Data` → `types/custom/Data`)
- Edge cases (null schemas, empty keys, trailing slashes)

## Benefits

1. **Full CIP-57 Compliance**: Handles optional title fields per specification
2. **Better Error Messages**: Descriptive messages with context about what went wrong
3. **Consistent Error Handling**: All blueprint processing errors use BlueprintGenerationException
4. **Clear Separation**: Blueprint processing errors vs runtime validation errors
5. **Fail Fast**: Compilation stops immediately with actionable errors
6. **Improved Test Coverage**: 22 new unit tests for edge cases

## Statistics

- **Files Created**: 7 (1 exception class, 1 test class, 5 test resources)
- **Files Modified**: 9
- **New Unit Tests**: 22
- **IllegalArgumentException → BlueprintGenerationException**: 6 replacements
- **Test Results**: ✅ All 213 tests passing

## Real-World Impact

This PR enables the annotation processor to handle real-world blueprints like SundaeSwap V2 that leverage CIP-57's optional title field. Previously, these blueprints would fail with NPE during compilation. Now they generate correct Java classes with proper error handling.

### CODE REVIEW

File: annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/BlueprintAnnotationProcessor.java [hunk 1]
```diff
@@ -132,8 +133,12 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                 }
 
                 String ns = getNamespaceFromReferenceKey(key);
-
-                fieldSpecProcessor.createDatumClass(ns, definition.getValue());
+                try {
+                    fieldSpecProcessor.createDatumClass(ns, key, definition.getValue());
+                } catch (BlueprintGenerationException e) {
```
Comment (matiwinnetou) [inline annotation-processor/src/main/java/com/bloxbean/cardano/client/plutus/annotation/processor/blueprint/BlueprintAnnotationProcessor.java]: TODO using exceptions for control flow, looks silly.
